src/server/pfs/server/driver_new_storage.go

```diff
@@ -333,7 +409,30 @@ func (d *driver) compact(master *work.Master, outputPath string, inputPrefixes [
 	}); err != nil {
 		return err
 	}
-	return d.storage.Compact(master.Ctx(), outputPath, []string{scratch})
+
+	var shardOutputs []string
```

Comment:
I think we should follow the same pattern you used for recording the `childOutputPaths` here. This does not actually preserve the shard order because walk will return the paths lexicographically.

Reply:
I think this can produce an error if no shard is written out.

Reply:
If a shard is created, then a fileset, empty or not, should get written to the semantic path for a shard (if it is not, then there is a bug somewhere).